utils/s3/retry_strategy.cc

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include "retry_strategy.hh"
+#include "aws_error.hh"
+
+using namespace std::chrono_literals;
+
+namespace aws {
+
+default_retry_strategy::default_retry_strategy(uint32_t max_retries, uint32_t scale_factor)
+    : _max_retries(max_retries)
+    , _scale_factor(scale_factor) {
+}
+
+bool default_retry_strategy::should_retry(const aws_error& error, uint32_t attempted_retries) const {
+    if (attempted_retries >= _max_retries) {
+        return false;
+    }
+
+    return error.is_retryable() == retryable::yes;
+}
+
+std::chrono::milliseconds default_retry_strategy::delay_before_retry(const aws_error&, uint32_t attempted_retries) const {
+    if (attempted_retries == 0) {
```

Comment:
It looks like this can never happen, caller first increases its `retries` counter, then calls this method

Reply:
changed caller behavior